argo-lite/src/services/AlgorithmUtils.js

```diff
@@ -44,4 +45,24 @@ export function averageClusteringCoefficient(snapshot) {
     const result = jsnx.averageClustering(jsnxGraph);
     console.log('Computing Clustering Coefficient');
     return result;
-}
\ No newline at end of file
+}
+
+/**
+ * Calculate the number of connected components in a graph
+ * @param {*} rawGraph the rawGraph inside appState
+ */
+export function connectedComponents(rawGraph) {
+    var idDict = {};
+    var i;
+    for (i = 0; i < rawGraph.nodes.length; i++) {
+        idDict[rawGraph.nodes[i].id] = i;
+    }
+    var g = new jsgraphs.Graph(rawGraph.nodes.length);
+    rawGraph.edges.forEach(e => {
```

Comment:
There is a `convertToJSGraph` function right above. It might be helpful to move this part of code to above to reuse them for other algorithms.

Reply:
Fixed in a new commit.